tests/tests_unit/test_data_classes/test_functions.py

```diff
@@ -3,17 +3,24 @@
 
 import pytest
 
+from cognite.client import CogniteClient
 from cognite.client.data_classes import Function, FunctionSchedule
 from tests.utils import jsgz_load
 
 
 @pytest.fixture
-def empty_function():
-    return Function(id=123, cognite_client=MagicMock())
+def mock_client():
+    (mock_client := MagicMock()).__class__ = CogniteClient
```

Comment:
`MagicMock(spec=CogniteClient)` isn't sufficient?

Reply:
:hide:

Reply:
`MagicMock(spec=CogniteClient)` broke 10 tests... hmm

Reply:
It can't inspect attributes that we set in `__init__` so it doesn't allow i.e. `client.functions`.